scipy/optimize/_qap.py

```diff
@@ -0,0 +1,389 @@
+        raise ValueError(msg)
+    elif seed_cost.shape[0] != seed_dist.shape[0]:
+        msg = "Seed arrays must be of equal size"
+        raise ValueError(msg)
+    elif seed_cost.shape[0] > cost_matrix.shape[0]:
+        msg = "There cannot be more seeds than there are nodes"
+        raise ValueError(msg)
+    elif not (seed_cost >= 0).all() or not (seed_dist >= 0).all():
+        msg = "Seed array entries must be greater than or equal to zero"
+        raise ValueError(msg)
+    elif (
+        not (seed_cost <= (cost_matrix.shape[0] - 1)).all()
+        or not (seed_dist <= (cost_matrix.shape[0] - 1)).all()
+    ):
+        msg = "Seed array entries must be less than or equal to n-1"
+        raise ValueError(msg)
+    elif type(n_init) is not int and n_init <= 0:
+        msg = '"n_init" must be a positive integer'
+        raise TypeError(msg)
+    elif not init_method == "barycenter" and not init_method == "rand":
+        msg = 'Invalid "init_method" parameter string'
+        raise ValueError(msg)
+    elif max_iter <= 0 and type(max_iter) is not int:
+        msg = '"max_iter" must be a positive integer'
+        raise TypeError(msg)
+    elif type(shuffle_input) is not bool:
+        msg = '"shuffle_input" must be a boolean'
+        raise TypeError(msg)
+    elif eps <= 0 and type(eps) is not float:
+        msg = '"eps" must be a positive float'
+        raise TypeError(msg)
+    elif type(maximize) is not bool:
+        msg = '"maximize" must be a boolean'
+        raise TypeError(msg)
+
+    n = cost_matrix.shape[0]  # number of vertices in graphs
+    n_seeds = seed_cost.shape[0]  # number of seeds
+    n_unseed = n - n_seeds
+
+    score = math.inf
+    perm_inds = np.zeros(n)
+
+    obj_func_scalar = 1
+    if maximize:
+        obj_func_scalar = -1
+        score = 0
+
+    seed_dist_c = np.setdiff1d(range(n), seed_dist)
+    if shuffle_input:
+        seed_dist_c = np.random.permutation(seed_dist_c)
+        # shuffle_input to avoid results from inputs that were already matched
+
+    seed_cost_c = np.setdiff1d(range(n), seed_cost)
+    permutation_cost = np.concatenate([seed_cost, seed_cost_c],
+                                      axis=None).astype(int)
+    permutation_dist = np.concatenate([seed_dist, seed_dist_c],
+                                      axis=None).astype(int)
+    cost_matrix = cost_matrix[np.ix_(permutation_cost, permutation_cost)]
+    dist_matrix = dist_matrix[np.ix_(permutation_dist, permutation_dist)]
+
+    # definitions according to Seeded Graph Matching [2].
+    A11 = cost_matrix[:n_seeds, :n_seeds]
+    A12 = cost_matrix[:n_seeds, n_seeds:]
+    A21 = cost_matrix[n_seeds:, :n_seeds]
+    A22 = cost_matrix[n_seeds:, n_seeds:]
+    B11 = dist_matrix[:n_seeds, :n_seeds]
+    B12 = dist_matrix[:n_seeds, n_seeds:]
+    B21 = dist_matrix[n_seeds:, :n_seeds]
+    B22 = dist_matrix[n_seeds:, n_seeds:]
+    A11T = np.transpose(A11)
```

Comment:
Can you think of a case where this will not simply return a view? If not, consider simply `A11.T` as needed rather than pre-defining these variables.